### PR TITLE
Document most of OpenStackTaster in YARD

### PR DESCRIPTION
**Pull request by tolvstaa**

### CODE REVIEW

File: lib/openstack_taster.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@
 require 'net/ssh'
 require 'pry'
 
+# @author Andrew Tolvstad, Samarendra Hedaoo
```
Comment (Codysseus) [inline lib/openstack_taster.rb:9]: Add Cody Holliday

File: lib/openstack_taster.rb [hunk 2]
```diff
@@ -147,6 +152,10 @@ def taste(image)
     end
   end
 
+  # Write an error message to the log and optionally stdout.
+  # @param filename [String] the file in the log directory to write
+  # @param message [String] the message to write
+  # @param dup_stdout [Boolean] whether or not to print the message to stdout
   def error_log(filename, message, dup_stdout = false)
```
Comment (Codysseus) [inline lib/openstack_taster.rb:159]: Parameters are reworked.

File: lib/openstack_taster.rb [hunk 3]
```diff
@@ -176,6 +192,11 @@ def create_image(instance)
       .wait_for { status == 'active' }
   end
 
+  # Run the set of tests for each available volume on an instance.
+  # @param instance [Fog::Compute::OpenStack::Server] the instance to query
+  # @param username [String] the username to use when logging into the instance
+  # @param ssh_logger [Logger] the logger object for logging the SSH connection
+  # @return [Boolean] Whether or not the tests succeeded
   def test_volumes(instance, username, ssh_logger)
```
Comment (Codysseus) [inline lib/openstack_taster.rb:200]: No more SSH logger.

File: lib/openstack_taster.rb [hunk 4]
```diff
@@ -211,6 +232,11 @@ def test_volumes(instance, username, ssh_logger)
     end
   end
 
+  # A helper method to execute a series of commands remotely on an instance. This helper
+  #   passes its block directly to `Net::SSH#start()`.
+  # @param instance [Fog::Compute::OpenStack::Server] the instance on which to run the commands
+  # @param username [String] the username to use when logging into the instance
+  # @param ssh_logger [Logger] the logger object for logging the SSH connection
   def with_ssh(instance, username, ssh_logger, &block)
```
Comment (Codysseus) [inline lib/openstack_taster.rb:240]: No more SSH logger.

File: lib/openstack_taster.rb [hunk 5]
```diff
@@ -264,6 +294,12 @@ def volume_attach?(instance, volume)
     false
   end
 
+  # Test volume mounting and unmounting for an instance and a volume.
+  # @param instance [Fog::Compute::OpenStack::Server] the instance on which to mount the volume
+  # @param username [String] the username to use when logging into the instance
+  # @param ssh_logger [Logger] the logger object for logging the SSH connection
+  # @param volume [Fog::Volume::OpenStack::Volume] the volume to mount
+  # @return [Boolean] whether or not the mounting/unmounting was successful
   def volume_mount_unmount?(instance, username, ssh_logger, volume)
```
Comment (Codysseus) [inline lib/openstack_taster.rb:303]: No more SSH logger.

File: lib/openstack_taster.rb [hunk 6]
```diff
@@ -304,6 +340,10 @@ def volume_mount_unmount?(instance, username, ssh_logger, volume)
     true
   end
 
+  # Log instance's partition listing.
+  # @param instance [Fog::Compute::OpenStack::Server] the instance to log
+  # @param username [String] the username to use when logging in to the instance
+  # @param ssh_logger [Logger] the logger object for logging the SSH connection
   def log_partitions(instance, username, ssh_logger)
```
Comment (Codysseus) [inline lib/openstack_taster.rb:347]: No more SSH logger

**Review by Codysseus — APPROVED** [review]
After the logger PR and the security PR are merged this will have to be reworked, but I can do that later. I left notes on each of the lines to be changed as a reminder, so disregard them.

**Review by knightsamar — COMMENTED** [review]
We need to fix the issue with the "missing first letter problem"

**Comment by Codysseus** [comment]
@knightsamar What is the "missing first letter" problem?

**Comment by knightsamar** [comment]
@DesertBeagle If you generate the HTML docs, you will see that the first letter of every function name and first parameter is missing.

**Comment by Codysseus** [comment]
@knightsamar I played around and found that the parser cuts off the first letter and bases the formatting on the code rather than the documentation comments.

**Comment by Codysseus** [comment]
Closing pull request since this PR diverges so greatly from master.

**Pull request closed without merging**